libdnf/base/goal.cpp

```diff
@@ -774,6 +830,13 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
 
     auto sack = base->get_rpm_package_sack();
     rpm::PackageQuery base_query(base);
+
+    // Apply advisory filters
+    if (settings.advisory_filter.has_value()) {
```

Comment:
I believe that security filters must be applied after repoid filters special for upgrade minimal where eq or the first higher is used but the first higher can be from another repository or a different vendor.